src/backend/marsha/core/api/video.py

```diff
@@ -820,3 +825,129 @@ def initiate_upload(self, request, pk=None):
         Thumbnail.objects.filter(pk=pk).update(upload_state=defaults.PENDING)
 
         return Response(presigned_post)
+
+
+class SharedLiveMediaViewSet(ObjectPkMixin, viewsets.ModelViewSet):
+    """Viewset for the API of the SharedLiveMedia object."""
+
+    permission_classes = [permissions.NotAllowed]
+    queryset = SharedLiveMedia.objects.all()
+    serializer_class = serializers.SharedLiveMediaSerializer
+
+    def get_serializer_context(self):
+        """Extra context provided to the serializer class."""
+        context = super().get_serializer_context()
+
+        user = self.request.user
+        # If the user is a JWT token and has roles check if roles are admin or instructor
+        if (
+            isinstance(user, TokenUser)
+            and user.token.get("roles")
+            and (
+                bool(LTI_ROLES[ADMINISTRATOR] & set(user.token.get("roles")))
+                or bool(LTI_ROLES[INSTRUCTOR] & set(user.token.get("roles")))
+            )
+        ):
+            context.update({"is_admin": True})
+
+        return context
+
+    def get_permissions(self):
+        """Instantiate and return the list of permissions that this view requires."""
+        if self.action in ["create", "list"]:
+            permission_classes = [
+                permissions.IsTokenInstructor
+                | permissions.IsTokenAdmin
+                | permissions.IsParamsVideoAdminThroughOrganization
+                | permissions.IsParamsVideoAdminThroughPlaylist
+            ]
+        elif self.action in ["retrieve"]:
+            permission_classes = [
+                permissions.IsTokenResourceRouteObjectRelatedVideo
+                | permissions.IsRelatedVideoPlaylistAdmin
+                | permissions.IsRelatedVideoOrganizationAdmin
+            ]
+        else:
+            permission_classes = [
+                permissions.IsTokenResourceRouteObjectRelatedVideo
+                & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+                | permissions.IsRelatedVideoPlaylistAdmin
+                | permissions.IsRelatedVideoOrganizationAdmin
+            ]
+        return [permission() for permission in permission_classes]
+
+    def list(self, request, *args, **kwargs):
+        """List shared live media through the API."""
+        queryset = self.get_queryset().none()
+        # If the "user" is just representing a resource and not an actual user profile,
+        # restrict the queryset to tracks linked to said resource
+        user = self.request.user
+        if isinstance(user, TokenUser) and (
+            not user.token.get("user")
+            or user.token.get("user", {}).get("id") != user.token.get("resource_id")
+        ):
+            queryset = (
+                self.get_queryset().filter(video__id=user.id).order_by("created_on")
+            )
+
```

Comment:
I don't think you suggestion allow to remove the `if` statement. Because the `video` can be in the query string while we are using a `JWT` token related to a video.

Reply:
If there is no `video` in the query string and no JWT token used, then we want to return no result.

Reply:
ping @sampaccoud

Reply:
ok ok.